raiden_contracts/utils/transaction.py

```diff
@@ -1,25 +1,38 @@
+from typing import Tuple
+
 from web3 import Web3
 from web3.utils.threads import (
     Timeout,
 )
 
 
-def check_succesful_tx(web3: Web3, txid: str, timeout=180) -> dict:
+def check_succesful_tx(web3: Web3, txid: str, timeout=180) -> Tuple[dict, dict]:
     '''See if transaction went through (Solidity code did not throw).
-    :return: Transaction receipt
+    :return: Transaction receipt and transaction info
     '''
     receipt = wait_for_transaction_receipt(web3, txid, timeout=timeout)
     txinfo = web3.eth.getTransaction(txid)
     if receipt['status'] == 0:
         raise ValueError(f"Status 0 indicates failure")
     if txinfo['gas'] == receipt['gasUsed']:
         raise ValueError(f"Gas is completely used ({txinfo['gas']}). Failure?")
-    return receipt
+    return (receipt, txinfo)
 
 
 def wait_for_transaction_receipt(web3, txid, timeout=180):
+    receipt = None
     with Timeout(timeout) as time:
-            while not web3.eth.getTransactionReceipt(txid):
+            while not receipt or not receipt['blockNumber']:
+                print('current receipt:')
+                print(receipt)
+                print('')
```

Comment:
Consider using the `logging` module.

Reply:
It was a debugging info. Removed from the commit.